software/webapp/src/main/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppSshDriver.java

```diff
@@ -103,6 +103,7 @@ public void install() {
                         BashCommands.installPackage("software-properties-common python-software-properties python g++ make"),
                         BashCommands.sudo("add-apt-repository ppa:chris-lea/node.js"))))
                 .add(BashCommands.installPackage(MutableMap.of("yum", "git nodejs npm", "apt", "git-core nodejs"), null))
+                .add("npm install n") // Create ~/.npm as the current user instead of root
```

Comment:
If it's just to create the folder, then create it explicitly, instead of installing the package locally and globally twice.